hls4ml/backends/fpga/fpga_types.py

```diff
@@ -233,6 +233,11 @@ def definition_cpp(self, name_suffix='', as_reference=False):
             type=self.type.name, name=self.name, suffix=name_suffix, shape=self.size_cpp(), pragma=self.pragma
         )
 
+class CatapultArrayVariableDefinition(VariableDefinition):
```

Comment:
I would prefer all these additions to not be placed here but in backends/catatapult/catapult_types.py. (The quartus and vivado additions should also be moved to corresponding files.)

Reply:
(Moving the quartus and vivado additions is outside the scope of this PR, though; this PR is just for catapult.)